server/models/Transaction.ts

```diff
@@ -1111,9 +1111,12 @@ Transaction.createPaymentProcessorFeeTransactions = async (
   const paymentMethod =
     transaction.PaymentMethodId && (await models.PaymentMethod.findByPk(transaction.PaymentMethodId));
   const payoutMethod = transaction.PayoutMethodId && (await models.PayoutMethod.findByPk(transaction.PayoutMethodId));
-  const vendor = await Transaction.getPaymentProcessorFeeVendor(
-    paymentMethod?.service || payoutMethod?.type || 'OTHER',
-  );
+  let paymentMethodService = paymentMethod?.service || payoutMethod?.type || 'OTHER';
+  if (paymentMethodService === PayoutMethodTypes.BANK_ACCOUNT && !data?.transfer) {
```

Comment:
Are we sure the PayoutMethodId is consistent across all transaction kinds that are generated?